flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/plan/batch/sql/CalcTest.scala

```diff
@@ -207,4 +210,19 @@ class CalcTest extends TableTestBase {
     val sqlQuery = "SELECT a FROM (SELECT a, b FROM MyTable) t WHERE random_udf(b) > 10"
     util.verifyRelPlan(sqlQuery)
   }
+
+  @Test
+  def testCalcMergeWithNonDeterministicExpr3(): Unit = {
```

Comment:
We can use a minimal case to reproduce the error, e.g.,
```scala
  @Test
  def testCalcMergeWithCorrelate(): Unit = {
    util.addTemporarySystemFunction("str_split", new StringSplit())
    val sqlQuery =
      """
        |
        |SELECT a, r FROM (
        | SELECT a, random_udf(b) r FROM (
        |  select a, b, c1 FROM MyTable, LATERAL TABLE(str_split(c)) AS T(c1)
        | ) t
        |)
        |WHERE r > 10
        |""".stripMargin
    util.verifyRelPlan(sqlQuery)
  }
```

Reply:
OK. The code and commit msg has been updated. Could you please check it again? Thanks~